cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -168,33 +191,31 @@ func checkPullRepositoryConfiguration(options abapEnvironmentPullGitRepoOptions)
 	return nil
 }
 
-func handlePull(repo abaputils.Repository, pullConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (err error) {
-	startPullLogs(repo.Branch)
-
-	uriConnectionDetails, err := triggerPull(repo.Name, pullConnectionDetails, client)
-	if err != nil {
-		return errors.Wrapf(err, "Pull of '%s' failed on the ABAP System", repo.Name)
-	}
-
-	// Polling the status of the repository import on the ABAP Environment system
-	status, errorPollEntity := abaputils.PollEntity(repo.Name, uriConnectionDetails, client, pollIntervall)
-	if errorPollEntity != nil {
-		return errors.Wrapf(errorPollEntity, "Pull of '%s' failed on the ABAP System", repo.Name)
+func startPullLogs(repo abaputils.Repository) {
+	commitString := ""
+	if repo.CommitID != "" {
+		commitString = ", commit '" + repo.CommitID + "'"
 	}
-	if status == "E" {
-		return errors.New("Pull of " + repo.Name + " failed on the ABAP System")
-	}
-	log.Entry().Info(repo.Name + " was pulled successfully")
-	return err
-}
-
-func startPullLogs(repositoryName string) {
 	log.Entry().Info("-------------------------")
-	log.Entry().Info("Start pulling " + repositoryName)
+	log.Entry().Info("Start pulling " + repo.Name + commitString)
 	log.Entry().Info("-------------------------")
 }
 
 func finishPullLogs() {
 	log.Entry().Info("-------------------------")
 	log.Entry().Info("All repositories were pulled successfully")
 }
+
+func convertPullConfig(config *abapEnvironmentPullGitRepoOptions) abaputils.AbapEnvironmentOptions {
+	subOptions := abaputils.AbapEnvironmentOptions{}
+
+	subOptions.CfAPIEndpoint = config.CfAPIEndpoint
+	subOptions.CfServiceInstance = config.CfServiceInstance
+	subOptions.CfServiceKeyName = config.CfServiceKeyName
+	subOptions.CfOrg = config.CfOrg
+	subOptions.CfSpace = config.CfSpace
+	subOptions.Host = config.Host
+	subOptions.Password = config.Password
+	subOptions.Username = config.Username
+	return subOptions
+}
```

Comment:
After seeing this 3 times: Should we also refactor this into the abaputils pkg?

Reply:
This is a type conversion with different types each time. I'm not sure how we could refactor this to make it easier. (If you have a suggestion, please let me know)

Reply:
The base line that should be working for all types should be the `abaputils.AbapEnvironmentOptions{}` object. What came to my mind is the following implementation (it's just a draft):
```
func convertConfig(config1 *abapEnvironmentPullGitRepoOptions, config2 *abapEnvironmentCheckoutBranchOptions, config3 *abapEnvironmentCloneGitRepoOptions) abaputils.AbapEnvironmentOptions {
	subOptions := abaputils.AbapEnvironmentOptions{}
        
        if (config1 != reflect.deepEqual(config, abapEnvironmentPullGitRepoOptions{}) {
	  subOptions.CfAPIEndpoint = config1.CfAPIEndpoint
	  subOptions.CfServiceInstance = config1.CfServiceInstance
	  subOptions.CfServiceKeyName = config1.CfServiceKeyName
	  subOptions.CfOrg = config1.CfOrg
	  subOptions.CfSpace = config1.CfSpace
	  subOptions.Host = config1.Host
	  subOptions.Password = config1.Password
	  subOptions.Username = config1.Username
        }

        /*  Check for the other config objects   */

	return subOptions
}
```

Optionally you could also provide a list of all step parameter types to a `convertConfig` function and loop over all configs and simply assigning the values to the `abaputils.AbapEnvironmentOptions{}` as outlined above.

I believe this is currently implemented in 3/4 steps separately which seems fine to me for the time being. This is just an idea.

Reply:
Thanks for the suggestion. But, I don't see the real benefit of this approach. In the end, all assignments are still there - only in one function. And this function seems to be (relatively) complicated:

- 3 input parameters, of which 2 are always empty
- if statements to determine, which one is not empty

-> Also, there is no disadvantage in having three functions with their respective purpose (and benefiting from easier function signatures and avoiding unnecessary if statements)